nifi-stateless/nifi-stateless-api/src/test/java/org/apache/nifi/stateless/config/PropertiesFileEngineConfigurationParserTest.java

```diff
@@ -0,0 +1,107 @@
+import org.junit.jupiter.api.AfterAll;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+import java.io.File;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.io.OutputStream;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.Properties;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+
+public class PropertiesFileEngineConfigurationParserTest {
+    private PropertiesFileEngineConfigurationParser parser;
+
+    private static Path narDirectory;
+
+    private static Path workingDirectory;
+
+    @BeforeAll
+    public static void setDirectories() throws IOException {
+        narDirectory = Files.createTempDirectory(PropertiesFileEngineConfigurationParserTest.class.getSimpleName());
+        workingDirectory = Files.createTempDirectory(PropertiesFileEngineConfigurationParserTest.class.getSimpleName());
+    }
+
+    @AfterAll
+    public static void deleteDirectories() throws IOException {
+        Files.deleteIfExists(narDirectory);
+        Files.deleteIfExists(workingDirectory);
+    }
+
+    @BeforeEach
+    public void setParser() {
+        parser = new PropertiesFileEngineConfigurationParser();
+    }
+
+    @Test
+    public void testParseEngineConfigurationRequiredProperties() throws IOException, StatelessConfigurationException {
+        final Properties properties = getRequiredProperties();
+        final File propertiesFile = getPropertiesFile(properties);
+
+        final StatelessEngineConfiguration configuration = parser.parseEngineConfiguration(propertiesFile);
+        assertNotNull(configuration);
+        assertEquals(narDirectory.toFile(), configuration.getNarDirectory());
+        assertEquals(workingDirectory.toFile(), configuration.getWorkingDirectory());
+    }
+
+    @Test
+    public void testParseEngineConfigurationRandomSensitivePropsKey() throws IOException, StatelessConfigurationException {
+        final Properties properties = getRequiredProperties();
+        final File propertiesFile = getPropertiesFile(properties);
+
+        final StatelessEngineConfiguration configuration = parser.parseEngineConfiguration(propertiesFile);
+        assertNotNull(configuration);
+
+        final String sensitivePropsKey = configuration.getSensitivePropsKey();
+        assertNotNull(sensitivePropsKey);
+        assertFalse(sensitivePropsKey.isEmpty());
+
+        final StatelessEngineConfiguration reloadedConfiguration = parser.parseEngineConfiguration(propertiesFile);
+        assertEquals(sensitivePropsKey, reloadedConfiguration.getSensitivePropsKey());
+    }
+
+    private Properties getRequiredProperties() {
+        final Properties properties = new Properties();
+
+        properties.setProperty("nifi.stateless.nar.directory", narDirectory.toString());
```

Comment:
Are these the only properties required of stateless?
(Is there a separate admin guide specifically for stateless?)

Reply:
These are the only properties currently required for the stateless parser to return a configuration. Several other properties fall back to default values, so as long as these two properties are present, and the directories exist, the parser returns a working configuration.

I am not aware additional documentation for Stateless properties, so that would be great to address in a separate issue.

Reply:
@markobean The [README.md](https://github.com/apache/nifi/blob/main/nifi-stateless/nifi-stateless-assembly/README.md) for Stateless provides current documentation on properties. Since it is still a work-in-progress, the details should be promoted to standard documentation in the future.